backend/src/ee/services/external-kms/external-kms-service.ts

```diff
@@ -51,15 +57,25 @@ export const externalKmsServiceFactory = ({
       actorAuthMethod,
       actorOrgId
     );
-    ForbiddenError.from(permission).throwUnlessCan(OrgPermissionActions.Edit, OrgPermissionSubjects.Settings);
+
+    ForbiddenError.from(permission).throwUnlessCan(OrgPermissionActions.Create, OrgPermissionSubjects.Kms);
+    const plan = await licenseService.getPlan(actorOrgId);
+    if (!plan.externalKms) {
+      throw new BadRequestError({
+        message: "Failed to create external KMS due to plan restriction. Upgrade to the Enterprise plan."
+      });
+    }
+
     const kmsSlug = slug ? slugify(slug) : slugify(alphaNumericNanoId(8).toLowerCase());
 
     let sanitizedProviderInput = "";
     switch (provider.type) {
       case KmsProviders.Aws:
         {
           const externalKms = await AwsKmsProviderFactory({ inputs: provider.inputs });
-          await externalKms.validateConnection();
+          if (provider.inputs.kmsKeyId) {
+            await externalKms.validateConnection();
```

Comment:
validate connection must always happen. Any reason for moving it like this?

Reply:
Validate connection fails when kmsKeyId is not provided by the user 
<img width="627" alt="image" src="https://github.com/user-attachments/assets/fcb0765c-f124-4f79-be1a-16c713a6dc6e">

Reply:
we're only creating the aws key if it does not exist below the validateConnection call
<img width="763" alt="image" src="https://github.com/user-attachments/assets/b6155257-e82b-4034-8f31-2e4e1428a015">